src/metabase/models/pulse.clj

```diff
@@ -310,11 +311,13 @@
                   :where  [:and
                            [:not= :p.alert_condition nil]
                            [:= :pc.card_id card-id]
-                           [:= :pcr.user_id user-id]]})))
+                           [:= :pcr.user_id user-id]
+                           [:= :p.archived archived?]]})))
 
 (defn retrieve-alerts-for-cards
   "Find all alerts for `card-ids`, used for admin users"
-  [& card-ids]
+  [{:keys [archived? card-ids]
+    :or   {archived? false}}]
```

Comment:
Same thing here, if someone does something like 

```clj
(retrieve-alerts-for-cards {:archived? (some-> archived-param-str Boolean/parseBoolean)})
```

and passes in `{:archived? nil}` then the query wouldn't return anything at all. So we should rework this a tiny bit